base-ci-resource.yaml

```diff
@@ -47,6 +69,9 @@ testNodeApplication:
   only:
     variables:       
       - $NODEPOSTGRESAPP == null  
+  except:
+    variables:
+      - $PLATFORM == "java"
```

Comment:
Maybe rather than an "except java" this should be an "only node". All our projects that need to define the platform, and if they don't I dont mind changing it.

Reply:
sure, I was worried about PLATFORM not being defined for all dependent projects.

Reply:
Yep, and I get that, and it makes sense. But its ok to do the right thing here, because if we don't, and we want to add another language, this will get ugly fast.  I'm ok paying a small tax if we've done it wrong where we've included this file, and I dont think we have anyway.

Reply:
Are you ok paying that tax immediately? We can't merge it until we pay the tax.

This is the downside of an unversioned build script.

Reply:
I think the way I have it now is like an or statement based on my testing ? Does this match with your suggestion?

Reply:
It is an OR statement, yes.

Hmm, but does that work? It would mean if we set PLATFORM to node, this step will run regardless of the POSTGRES setting, meaning that a configuration with NODEPOSTGRESAPP set to true and PLATFORM set to node will run both tests, and one will fail (because it needs the db set up)

Reply:
I think also it will make all node apps run testNodeApplication and testNodeApplicationUsingPostgres if I keep both variables under the only section

Reply:
Would need only NODE AND NOT POSTGRESAPP, right? I'm not sure if that is doable.